### PR TITLE
Run StdLib tests on sparkcore's MapFunc implementation

### PR DESCRIPTION
**Pull request by mossprescott**
This runs all the tests, skipping the Extract functions and the tests for `Power` since the implementation is limited to small values. 

Other than that, it found only a few small problems which are fixed here.

I tried to factor out the parts that should be common to QScript/MapFunc implementations, and included a few TODO comments about how that might work.

### CODE REVIEW

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/CoreMap.scala [hunk 1]
```diff
@@ -352,19 +353,19 @@ object CoreMap extends Serializable {
 
   private def between(d1: Data, d2: Data, d3: Data): Data = (d1, d2, d3) match {
     case (Data.Int(a), Data.Int(b), Data.Int(c)) =>
-      Data.Bool(a <= b && b <= c)
+      Data.Bool(b <= a && a <= c)
```
Comment (rabbitonweb) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/CoreMap.scala]: Then it means that function semantics were `a b c   =>     b  a  c` - wow, that was confusing! good you wrote those tests!

Comment (mossprescott) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/CoreMap.scala]: Yes, it's mirroring SQL's syntax: `a between b and c`.

**Comment by mossprescott** [comment]
@rabbitonweb can you go ahead and merge?

**Comment by mossprescott** [comment]
Merged latest from master. Will ping you when it's built.

**Pull request merged**